Sources/Update/update().swift

```diff
@@ -9,18 +9,62 @@
 */
 
 import struct PackageDescription.Version
+import Utility
+import POSIX
+import Get
 
+enum Error: ErrorProtocol {
+    case NoRepo(String)
+    case NoVersion(String)
+    case NoPackageName(String)
+}
 
-public protocol Package {
-    var path: String { get }
-    var version: Version { get }
-    var url: String { get }
+public struct Delta<T> {
+    public var added: [T] = []
+    public var removed: [T] = []
+    public var upgraded: [T] = []
+    public var downgraded: [T] = []
+}
 
-    func fetch()
-    func commit(newVersion: Version)
+public func update(root: String) throws {
+    for path in walk(root, recursively: false) {
+        guard path.isDirectory else { continue }
+        guard let repo = Git.Repo(path: path) else { throw Error.NoRepo(path) }
+        let oldVersion = repo.version
+        try repo.fetch()
+        try repo.upgrade()
+        print("updated:", try repo.pkgname(), oldVersion, "->", repo.version)
+    }
 }
 
+extension Git.Repo {
+    func upgrade() throws {
+        guard let latestVersion = versions.last else { throw Error.NoVersion(path) }
+        let vstr = (versionsArePrefixed ? "v" : "") + latestVersion.description
```

Comment:
It doesn't seem to me we should need to have this logic spread around the code. I would expect that there is a single place which loads a repository and determines what `Version` objects it has available, and also records how each of those is spelled. Without that it will be really hard to refactor the code to support new version scheme spellings.